internal/services/devtestlabs/dev_test_lab_resource.go

```diff
@@ -184,7 +185,13 @@ func resourceDevTestLabRead(d *pluginsdk.ResourceData, meta interface{}) error {
 		d.Set("artifacts_storage_account_id", props.ArtifactsStorageAccount)
 		d.Set("default_storage_account_id", props.DefaultStorageAccount)
 		d.Set("default_premium_storage_account_id", props.DefaultPremiumStorageAccount)
-		d.Set("key_vault_id", props.VaultName)
+
+		// TODO: remove the replacement until the issue(https://github.com/Azure/azure-rest-api-specs/issues/17422) is fixed.
+		// The lowercase "resourcegroups" and "microsoft.keyvault" in key value id are returned by Labs_Get API.
+		// This will cause the resource which referencing the key vault id to be re-created after running terraform apply command even though nothing has changed.
+		// Hence replace them as a workaround in terraform.
+		vaultId := strings.Replace(*props.VaultName, "resourcegroups", "resourceGroups", 1)
+		d.Set("key_vault_id", strings.Replace(vaultId, "microsoft.keyvault", "Microsoft.KeyVault", 1))
```

Comment:
Can we instead parse the Key Vault ID using the [`VaultID()`](https://github.com/hashicorp/terraform-provider-azurerm/blob/7c5ecd325127d5848149bd85f48beb527ee1969f/internal/services/keyvault/parse/vault.go#L41), and then setting the `id.ID()` back to the `key_vault_id`?

Reply:
thank you, fixed.